torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -88,43 +91,43 @@ void fuseBroadcast(Block *b) {
     // Can't fuse into nodes that don't support broadcasting
     if (!isBroadcasting(n)) continue;
 
-    // If the node already broadcasts, can't "rebroadcast"
-    // TODO: Actually, maybe you can, if there is a broadcast for some
-    // dims, and then another broadcast for the rest.  But this will
-    // never happen in practice so I didn't implement it.
-    if (n->hasAttribute(attr::broadcast) && n->i(attr::broadcast)) continue;
     JIT_ASSERT(!n->hasAttribute(attr::axis));
 
-    auto input_index = n->inputs().size() - 1;
-    auto* rhs_expand = n->input(input_index)->node();
-
-    // The rhs_expand input isn't actually an expand, so no fusion available
-    // XXX: we can't use the ->matches(...) mechanism in here, because input nodes
-    //      have been
-    if (rhs_expand->kind() != aten::expand ||
-        rhs_expand->input(1)->node()->kind() != onnx::Constant ||
-        rhs_expand->input(2)->node()->kind() != onnx::Constant) {
-      continue;
+    size_t index_begin = n->inputs().size() - 2;
```

Comment:
Here's an alternate strategy that I think is more explicit: for every operator that supports broadcasting, record the input positions for which they support broadcasting. This feels a bit more robust than saying, "Well... most broadcasting things have two arguments, except for gemm, which only has one broadcasting argument."